### DISCUSSION TITLE
Scribbles algorithm

### DISCUSSION THREAD

**Discussion by hw-ju** [Project-MONAI/MONAILabel#1141] [Q&A]
Hi!
Just curious what algorithm is used behind the Scribbles feature. Is it deep learning based?
<img src="https://user-images.githubusercontent.com/90806807/201490299-ce2e5d39-ad5b-48b0-a5ab-fdcced4f1277.png" height="80">

**Comment by tangy5** [comment]
Hi @hw-ju , yes, it's deep learning and statistic based algorithm. It's a AI-assisted correction tool, the scribble-based interactive segmentation correction can allow you to set ROIs for foreground/background, so that can update inference and correct annotation.

**Reply by hw-ju** [reply]
Thanks for your prompt response @tangy5! 
Could you point me to maybe a publication/documentation on the Scribble algorithm? Just curious about the details of that algorithm?

**Reply by diazandr3s** [reply]
Good question, @hw-ju. 
You can find all the information and reference articles about the scribbles algorithm here: https://github.com/Project-MONAI/MONAILabel/wiki/Scribbles
From what I understand the implemented algorithm in MONAI Label is not a deep-learning model. It is rather a machine learning model that uses Gaussian mixture models or energy functions.
Hope that helps.

**Reply by hw-ju** [reply]
Thanks @diazandr3s! Very helpful!

**Reply by Zrrr1997** [reply]
If you are still interested in a paper behind such interactive statistical approaches - this is the seminal Graph Cut paper [link](https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=937505). Classical approaches utilized pixel-level features regarding appearance and spatial distance to separate foreground from background. Such is also the foundation for energy-based optimization methods, such as CRFs ([seminal paper](https://proceedings.neurips.cc/paper/2011/file/beda24c1e1b46055dff2c39c98fd6fc1-Paper.pdf)). Both papers are a great read, especially if you enjoy reading about classical non-deep learning methods.

**Reply by hw-ju** [reply]
@Zrrr1997 Thanks for pointing to the seminal papers!
